doc-surrealql_versioned_docs/version-latest/statements/delete.mdx

```diff
@@ -72,6 +90,54 @@ DELETE user WHERE interests CONTAINS 'reading' RETURN BEFORE;
 DELETE user WHERE interests CONTAINS 'reading' RETURN AFTER;
 ```
 
+An important point to know when using a `WHERE` clause is that it performs a check on the [truthiness](/docs/surrealql/datamodel/values#values-and-truthiness) of a value, namely whether a value exists and is not a default value like 0, an empty string, empty array, and so on.
+
+As such, the `DELETE` query below that only specifies `WHERE age` essentially evaluates to "WHERE age exists" and will delete every cat in the database with an `age`.
+
+```surql
+CREATE cat:one SET age = 4;
+CREATE cat:two;
+DELETE cat WHERE age;
+SELECT * FROM cat;
+```
+
+```bash title="Output"
+[
+	{
+		id: cat:two
+	}
+]
+```
+
+This pattern is particularly useful when using SurrealDB's [literal types](/docs/surrealql/datamodel/literals) which were added in version 2.0. A literal type containing objects that contain a single top-level field can easily be matched on through the field name.
+
+```surql
+DEFINE FIELD error_info ON TABLE info TYPE 
```

Comment:
question: I am tempted to display `|` before the type def like prettier does in TypeScript, like this:

```ts
type X = 
   { continue: { message: "Continue" } } 
  | { retry_with_id: { error: string  } }
  | { deprecated: { message: string  } };
```

In reality it's more like this, but I am not sure this syntax is currently supported:


```ts
type X = 
  | { continue: { message: "Continue" } } 
  | { retry_with_id: { error: string  } }
  | { deprecated: { message: string  } };
```

What do you think?

Reply:
Yep, definitely nicer. I changed the other two examples in the docs to have the bar on the left as well.